app/javascript/crayons/Pills/Pill.jsx

```diff
@@ -0,0 +1,93 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import classNames from 'classnames/bind';
+import { Icon } from '@crayons';
+import XIcon from '@images/x.svg';
+
+export const Pill = ({
+  children,
+  element = 'button',
+  iconLeft,
+  iconRight,
+  iconRightDestructive,
+  className,
+  tooltip,
+  onKeyUp,
+  ...otherProps
+}) => {
+  const Element = element;
+  const restOfProps =
+    element === 'button'
+      ? { type: 'button', onKeyUp: handleKeyUp }
+      : element === 'a'
+      ? { href: '#' }
+      : '';
+
+  const [suppressTooltip, setSuppressTooltip] = useState(false);
+
+  const handleKeyUp = (event) => {
+    onKeyUp?.(event);
+    if (!tooltip) {
+      return;
+    }
+    setSuppressTooltip(event.key === 'Escape');
+  };
```

Comment:
This needs to be declared before we use it on line 22 - at the minute the keyUp event is always `undefined`

Reply:
Fixed.